### PR TITLE
Doggy v2

### PR DESCRIPTION
**Pull request by bai**

### CODE REVIEW

File: lib/doggy/shared_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+          raise e if tries >= times && reraise
+        end
+      end
+    end
+
+    def self.agree(prompt)
+      raise Error, "Not a tty" unless $stdin.tty?
+
+      puts prompt + " (Y/N)"
```
Comment (kvs) [inline lib/doggy/shared_helpers.rb]: maybe `print prompt + " (Y/N) "`, so the cursor remains on the same line?

File: lib/doggy/model/screen.rb [hunk 2]
```diff
@@ -30,10 +16,20 @@ def self.upload(ids)
...
 
-    def self.create(name)
-      item = new(description: { 'board_title' => name, 'widgets' => [] })
-      item.push
-      item.save
+    def self.edit(id)
+      system %{open "https://app.datadoghq.com/screen/#{id}"}
+      if SharedHelpers.agree("Are you done?")
+        puts 'Here is the output of your edit:'
+        puts Doggy::Serializer::Json.dump(new(id: id).raw)
```
Comment (kvs) [inline lib/doggy/model/screen.rb]: Hmm, so screenboard edit doesn't update the local file?

File: lib/doggy/friendly_errors.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    puts e
+    exit 1
+  rescue SystemExit => e
+    exit e.status
+  rescue Exception => e
+    request_issue_report_for(e)
+    exit 1
+  end
+
+  def self.request_issue_report_for(e)
```
Comment (kvs) [inline lib/doggy/friendly_errors.rb]: Neat!

File: lib/doggy/shared_helpers.rb (hunk 1 above)
Comment (pallan) [inline lib/doggy/shared_helpers.rb]: why not `gets` ?

**Pull request merged**